tests/conftest.py

```diff
@@ -312,6 +314,8 @@ def stub_config():
     for key in default_keys:
         config.register(*key)
     config.extend(defaults.copy())
+    # Patch load() so we don't update any key/value pairs from actual files:
+    config.load = mock.Mock(side_effect=lambda: setattr(config, "ready", True))
```

Comment:
I tried to figure out a clean way to replace the configuration wholesale in tests, or ... something else better than what we have, and failed. In the end I just moved the patching of `load` to the `stub_config` fixture and simplified `active_config` slightly, but it's essentially the same as what you had before.